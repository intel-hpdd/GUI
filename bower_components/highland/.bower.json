{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}