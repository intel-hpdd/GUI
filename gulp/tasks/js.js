//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
