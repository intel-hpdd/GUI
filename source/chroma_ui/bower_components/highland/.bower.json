{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}