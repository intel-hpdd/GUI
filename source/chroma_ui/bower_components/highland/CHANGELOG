Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
