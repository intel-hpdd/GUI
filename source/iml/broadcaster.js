// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import highland from 'highland';

import type {
  HighlandStreamT,
  errorWrapT
} from 'highland';


type streamFnT = () => HighlandStreamT<any>;

export default function broadcaster (source$:HighlandStreamT<any>):streamFnT {
  var latest:mixed;

  const viewers = [];

  source$
    .errors(error => {
      const err:errorWrapT = {
        __HighlandStreamError__: true,
        error
      };

      viewers.forEach(v => v.write(err));
    })
    .each((xs:mixed) => {
      latest = xs;

      viewers.forEach(v => v.write(xs));
    });

  const fn = () => {
    const viewer$:HighlandStreamT<any> = highland()
      .onDestroy(() => {
        const idx = viewers
          .indexOf(viewer$);

        if (idx !== -1)
          viewers.splice(idx, 1);
      });

    if (latest)
      viewer$.write(latest);

    viewers.push(viewer$);

    return viewer$;
  };

  fn.endBroadcast = () => source$.destroy();

  return fn;
}
