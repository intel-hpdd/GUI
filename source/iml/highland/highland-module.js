// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
