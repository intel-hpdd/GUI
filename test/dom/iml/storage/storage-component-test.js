// @flow

import Inferno from 'inferno';
import highland, { type HighlandStreamT } from 'highland';
import broadcaster from '../../../../source/iml/broadcaster.js';
import type { State } from '../../../../source/iml/storage/storage-reducer.js';
import { renderToSnapshot } from '../../../test-utils.js';

describe('storage component', () => {
  let StorageComponent,
    mockStore,
    storage$: HighlandStreamT<State>,
    alertIndicator$: HighlandStreamT<Object[]>;

  beforeEach(() => {
    mockStore = { dispatch: jest.fn() };
    jest.mock('../../../../source/iml/store/get-store.js', () => mockStore);

    const mockStorageResources = jest.fn(() => highland([]));
    jest.mock(
      '../../../../source/iml/storage/storage-resources.js',
      () => mockStorageResources
    );

    ({
      StorageComponent
    } = require('../../../../source/iml/storage/storage-component.js'));

    storage$ = highland();
    alertIndicator$ = highland();
  });

  it('should render no plugins', () => {
    storage$.write({
      config: {
        sortDesc: false,
        sortKey: 'name',
        loading: false,
        selectIndex: 0,
        entries: 10,
        offset: 0
      },
      resourceClasses: [],
      resources: {
        meta: {
          limit: 10,
          next: null,
          offset: 0,
          previous: null,
          total_count: 0
        },
        objects: []
      }
    });

    expect(
      renderToSnapshot(
        <StorageComponent
          viewer={broadcaster(storage$)}
          alertIndicatorB={broadcaster(alertIndicator$)}
        />
      )
    ).toMatchSnapshot();
  });

  it('should render with plugins', () => {
    storage$.write({
      config: {
        sortDesc: false,
        sortKey: 'name',
        loading: true,
        selectIndex: 0,
        entries: 1,
        offset: 0
      },
      resourceClasses: [
        {
          class_name: 'EMCPower',
          columns: [
            {
              label: 'Size',
              name: 'size'
            },
            {
              label: 'Filesystem type',
              name: 'filesystem_type'
            },
            {
              label: 'Uuid',
              name: 'uuid'
            }
          ],
          fields: [
            {
              class: 'Bytes',
              label: 'Size',
              name: 'size',
              optional: false,
              user_read_only: false
            },
            {
              class: 'Boolean',
              label: 'Filesystem type',
              name: 'filesystem_type',
              optional: true,
              user_read_only: false
            },
            {
              class: 'String',
              label: 'Uuid',
              name: 'uuid',
              optional: false,
              user_read_only: false
            }
          ],
          id: 1,
          label: 'linux-EMCPower',
          modified_at: '2017-07-19T13:58:02.615410',
          plugin_internal: true,
          plugin_name: 'linux',
          resource_uri: '/api/storage_resource_class/1/',
          user_creatable: false
        }
      ],
      resources: {
        meta: {
          limit: 10,
          next: null,
          offset: 0,
          previous: null,
          total_count: 1
        },
        objects: []
      }
    });

    expect(
      renderToSnapshot(
        <StorageComponent
          viewer={broadcaster(storage$)}
          alertIndicatorB={broadcaster(alertIndicator$)}
        />
      )
    ).toMatchSnapshot();
  });
});
