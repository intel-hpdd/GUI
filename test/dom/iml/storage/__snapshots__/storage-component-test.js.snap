// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage component should render no plugins 1`] = `
<div>
  <div
    class="container container-full storage container"
  >
    <div
      class="well text-center no-plugins"
    >
      <h1>
        No storage plugins are currently installed.
      </h1>
      <p>
        When storage plugins are installed, use this tab to configure and view storage resources such as controllers.
      </p>
    </div>
  </div>
</div>
`;

exports[`storage component should render with plugins 1`] = `
<div>
  <div
    class="container container-full storage container"
  >
    <div>
      <h4
        class="section-header"
      >
        Storage Resources
      </h4>
      <div
        class="form-group well"
      >
        <label
          class="control-label"
        >
          Select Resource Class
        </label>
        <select
          class="form-control"
        >
          <option
            value="EMCPower,linux"
          >
            linux-EMCPower
          </option>
        </select>
      </div>
      <div
        class="table-container loading"
      >
        <div
          class="well text-center resource-table-no-records"
        >
          <h1>
            No records found
          </h1>
        </div>
        <div
          class="add-storage-device"
        >
          <h4
            class="section-header"
          >
            Add Storage Device
          </h4>
          <a
            class="btn btn-default btn-sm"
            href="http://localhost/configure/storage/add"
            type="button"
          >
            <i
              class="fa fa-plus-circle text-success"
            />
             Add Storage Device
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
